scripts/sct_label_utils.py

```diff
@@ -850,93 +225,185 @@ def main(args=None):
+        labels = arguments.create
+        out = sct_labels.create_labels_empty(img, labels)
     elif arguments.create_add is not None:
-        process_type = 'create-add'
-        input_coordinates = arguments.create_add
+        labels = arguments.create_add
+        out = sct_labels.create_labels(img, labels)
     elif arguments.create_seg is not None:
-        process_type = 'create-seg'
-        input_coordinates = arguments.create_seg
-    # TODO: This argument was not present in argparse. Should it be included?
-    # elif arguments.cross is not None:
-    #     process_type = 'cross'
-    #     input_cross_radius = arguments.cross
+        labels = []
+        for str_pair in arguments.create_seg:
+            z, z_val = str_pair.split(',')
+            labels.append(tuple([int(z), int(z_val)]))
+        out = sct_labels.create_labels_along_segmentation(img, labels)
     elif arguments.cubic_to_point:
-        process_type = 'cubic-to-point'
+        out = sct_labels.cubic_to_point(img)
     elif arguments.display:
-        process_type = 'display-voxel'
+        display_voxel(img, verbosity)
     elif arguments.increment:
-        process_type = 'increment'
+        out = sct_labels.increment_z_inverse(img)
     elif arguments.vert_body is not None:
-        process_type = 'vert-body'
-        vertebral_levels = arguments.vert_body
-    # elif '-vert-disc' in arguments:
-    #     process_type = 'vert-disc'
-    #     vertebral_levels = arguments['-vert-disc']
+        levels = arguments.vert_body
+        if len(levels) == 1 and levels[0] == 0:
+            levels = None # all levels
+        out = sct_labels.label_vertebrae(img, levels)
     elif arguments.vert_continuous:
-        process_type = 'vert-continuous'
+        out = sct_labels.continuous_vertebral_levels(img)
+        dtype = 'float32'
     elif arguments.MSE is not None:
-        process_type = 'MSE'
-        input_fname_ref = arguments.r
+        ref = Image(arguments.MSE)
+        mse = sct_labels.compute_mean_squared_error(img, ref)
+        sct.printv(f"Computed MSE: {mse}")
+        return
     elif arguments.remove_reference is not None:
-        process_type = 'remove-reference'
-        input_fname_ref = arguments.remove_reference
-    elif arguments.remove_symm is not None:
-        process_type = 'remove-symm'
-        input_fname_ref = arguments.r
-    elif arguments.create_viewer is not None:
-        process_type = 'create-viewer'
-        value = arguments.create_viewer
+        ref = Image(arguments.remove_reference)
+        out = sct_labels.remove_missing_labels(img, ref)
+    elif arguments.remove_sym is not None:
+        # first pass use img as source
+        ref = Image(arguments.remove_reference)
+        out = sct_labels.remove_missing_labels(img, ref)
+
+        # second pass use previous pass result as reference
+        ref_out = sct_labels.remove_missing_labels(ref, out)
+        ref_out.save(path=ref.absolutepath)
```

Comment:
Wow, you did a great job at simplifying the "remove labels" functions. This replaces a _lot_ of old code.

Reply:
:smile:  Yes it was a bit convoluted before.